Agent reports `panic: duplicate metrics collector registration attempted`

I am attempting to run `parca-agent` without any flags 0.32.0 on an ubuntu server.

After a few seconds it crashes:
```
./parca-agent
WARN[0000] Failed to setup gRPC connection (try 1 of 5): failed to exit idle mode: passthrough: received empty target in Build()
panic: duplicate metrics collector registration attempted

goroutine 1 [running]:
github.com/prometheus/client_golang/prometheus.(*Registry).MustRegister(0xc000d34aa0, {0xc00082a170?, 0x0?, 0x0?})
        github.com/prometheus/client_golang@v1.19.1/prometheus/registry.go:405 +0x66
github.com/parca-dev/parca-agent/flags.FlagsRemoteStore.setupGrpcConnection({{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, 0x0, 0x0, 0x2540be400, 0x0, ...}, ...)
        github.com/parca-dev/parca-agent/flags/grpc.go:110 +0x92b
github.com/parca-dev/parca-agent/flags.FlagsRemoteStore.WaitGrpcEndpoint({{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, 0x0, 0x0, 0x2540be400, 0x0, ...}, ...)
        github.com/parca-dev/parca-agent/flags/grpc.go:35 +0x148
main.mainWithExitCode()
        github.com/parca-dev/parca-agent/main.go:163 +0xa1b
main.main()
        github.com/parca-dev/parca-agent/main.go:94 +0x13
```

I'm observing the same. In my case it's happening when Parca server is not yet available (after restart) and Parca Agent(s) are trying to connect. When Parca server is up this is no longer happening. I'm running server in  K8s statefulSet and Agents daemonSet.
Parca v0.22
Parca Agent v0.32
  
```
time="2024-09-06T11:14:39Z" level=warning msg="Failed to setup gRPC connection (try 1 of 5): context deadline exceeded: connection error: desc = \"transport: Error while dialing: dial tcp 10.100.33.193:7070: connect: connection refused\""
panic: duplicate metrics collector registration attempted

goroutine 1 [running]:
github.com/prometheus/client_golang/prometheus.(*Registry).MustRegister(0xc00083f950, {0xc00086e760?, 0x0?, 0x0?})
        github.com/prometheus/client_golang@v1.19.1/prometheus/registry.go:405 +0x66
github.com/parca-dev/parca-agent/flags.FlagsRemoteStore.setupGrpcConnection({{0xc000a5c080, 0xa}, {0x0, 0x0}, {0x0, 0x0}, 0x1, 0x0, 0x2540be400, 0x0, ...}, ...)
        github.com/parca-dev/parca-agent/flags/grpc.go:110 +0x92b
github.com/parca-dev/parca-agent/flags.FlagsRemoteStore.WaitGrpcEndpoint({{0xc000a5c080, 0xa}, {0x0, 0x0}, {0x0, 0x0}, 0x1, 0x0, 0x2540be400, 0x0, ...}, ...)
        github.com/parca-dev/parca-agent/flags/grpc.go:35 +0x148
main.mainWithExitCode()
        github.com/parca-dev/parca-agent/main.go:163 +0xa1b
main.main()
        github.com/parca-dev/parca-agent/main.go:94 +0x13
```

--- a/flags/grpc.go
+++ b/flags/grpc.go
@@ -8,11 +8,11 @@ import (
 	"strings"
 	"time"
 
-	"github.com/open-telemetry/opentelemetry-ebpf-profiler/libpf"
 	grpc_prometheus "github.com/grpc-ecosystem/go-grpc-middleware/providers/prometheus"
 	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/logging"
 	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/retry"
 	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/timeout"
+	"github.com/open-telemetry/opentelemetry-ebpf-profiler/libpf"
 	"github.com/prometheus/client_golang/prometheus"
 	log "github.com/sirupsen/logrus"
 	tracing "go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
@@ -30,9 +30,20 @@ func (f FlagsRemoteStore) WaitGrpcEndpoint(ctx context.Context, reg prometheus.R
 	tick := time.NewTicker(libpf.AddJitter(f.GRPCStartupBackoffTime, 0.2))
 	defer tick.Stop()
 
+	// metrics
+	metrics := grpc_prometheus.NewClientMetrics(
+		grpc_prometheus.WithClientHandlingTimeHistogram(
+			grpc_prometheus.WithHistogramOpts(&prometheus.HistogramOpts{
+				NativeHistogramBucketFactor: 1.1,
+				Buckets:                     nil,
+			}),
+		),
+	)
+	reg.MustRegister(metrics)
+
 	var retries uint32
 	for {
-		if grpcConn, err := f.setupGrpcConnection(ctx, reg, tp); err != nil {
+		if grpcConn, err := f.setupGrpcConnection(ctx, metrics, tp); err != nil {
 			if retries >= f.GRPCMaxConnectionRetries {
 				return nil, err
 			}
@@ -57,7 +68,7 @@ func (f FlagsRemoteStore) WaitGrpcEndpoint(ctx context.Context, reg prometheus.R
 }
 
 // setupGrpcConnection sets up a gRPC connection instrumented with our auth interceptor
-func (f FlagsRemoteStore) setupGrpcConnection(parent context.Context, reg prometheus.Registerer, tp trace.TracerProvider) (*grpc.ClientConn, error) {
+func (f FlagsRemoteStore) setupGrpcConnection(parent context.Context, metrics *grpc_prometheus.ClientMetrics, tp trace.TracerProvider) (*grpc.ClientConn, error) {
 	encoding.RegisterCodec(vtprotoCodec{})
 
 	//nolint:staticcheck
@@ -97,17 +108,6 @@ func (f FlagsRemoteStore) setupGrpcConnection(parent context.Context, reg promet
 			NewPerRequestBearerToken(strings.TrimSpace(string(b)), f.Insecure)),
 		)
 	}
-
-	// metrics
-	metrics := grpc_prometheus.NewClientMetrics(
-		grpc_prometheus.WithClientHandlingTimeHistogram(
-			grpc_prometheus.WithHistogramOpts(&prometheus.HistogramOpts{
-				NativeHistogramBucketFactor: 1.1,
-				Buckets:                     nil,
-			}),
-		),
-	)
-	reg.MustRegister(metrics)
 
 	// tracing
 	exemplarFromContext := func(ctx context.Context) prometheus.Labels {

